src/Octoshift/Services/GithubApi.cs

```diff
@@ -1089,24 +1090,102 @@ public virtual async Task<string> UploadArchiveToGithubStorage(string orgDatabas
 
         if (isMultipart)
         {
-            var url = $"https://uploads.github.com/organizations/{orgDatabaseId.EscapeDataString()}/gei/archive/blobs/uploads";
+            var url = $"{_base_url}/{orgDatabaseId.EscapeDataString()}/gei/archive/blobs/uploads";
 
-#pragma warning disable IDE0028
-            using var multipartFormDataContent = new MultipartFormDataContent();
-            multipartFormDataContent.Add(streamContent, "archive", archiveName);
-#pragma warning restore
-
-            response = await _client.PostAsync(url, multipartFormDataContent);
+            response = await UploadMultipart(archiveContent, archiveName, url);
+            return response;
         }
         else
         {
-            var url = $"https://uploads.github.com/organizations/{orgDatabaseId.EscapeDataString()}/gei/archive?name={archiveName.EscapeDataString()}";
+            var url = $"{_base_url}/{orgDatabaseId.EscapeDataString()}/gei/archive?name={archiveName.EscapeDataString()}";
 
             response = await _client.PostAsync(url, streamContent);
+            var data = JObject.Parse(response);
+            return (string)data["uri"];
         }
+    }
 
-        var data = JObject.Parse(response);
-        return (string)data["uri"];
+    private async Task<string> UploadMultipart(Stream archiveContent, string archiveName, string uploadUrl)
+    {
+        var buffer = new byte[_streamSizeLimit];
+
+        // 1. Start the upload
+        var (startResponse, startHeaders) = await StartUploadAsync(uploadUrl, archiveName, archiveContent.Length);
+
+        var nextUrl = GetNextUrl(startHeaders);
+        var guid = System.Web.HttpUtility.ParseQueryString(nextUrl.Query)["guid"];
```

Comment:
I would add `System.Web` to the usings list to simplify this:
```suggestion
        var guid = HttpUtility.ParseQueryString(nextUrl.Query)["guid"];
```